### PR TITLE
VTQuery Filter Optimization

### PR DESCRIPTION
**Pull request by CyanRook**
* Replace std::map with std::unordered_map
* Moved using declarations to top
* Swapped copies for const references in function calls
* Moved filter check to last check before duplicate check

### CODE REVIEW

**Review by flippmoke — APPROVED** [review]

File: src/vtquery.cpp [hunk 1]
```diff
@@ -246,7 +248,7 @@ std::vector<vtzero::property> get_properties_vector(vtzero::feature& feat) {
     return v;
 }
 
-double convert_to_double(value_type value) {
+double convert_to_double(const value_type& value) {
```
Comment (flippmoke) [inline src/vtquery.cpp]: Nitpicky: I think we typically use `const&` style in this library?

**Comment by flippmoke** [comment]
> Moved filter check to last check before duplicate check

Can you provide a test case to show why this is important? Or perhaps just explain, curious whats the case here we are missing.

File: src/vtquery.cpp [hunk 2]
```diff
@@ -91,6 +91,8 @@ struct TileObject {
 };
 
 using value_type = boost::variant<float, double, int64_t, uint64_t, bool, std::string>;
+using key_type = std::string;
```
Comment (flippmoke) [inline src/vtquery.cpp]: Nitpicky: Why not just use `map_type::key_type` here?

**Comment by CyanRook** [comment]
> > Moved filter check to last check before duplicate check
> 
> Can you provide a test case to show why this is important?

We were seeing some slight performance issues and moved it down after some cheaper checks for skipping features.  It won't affect output, just how quickly we `continue` in the loop.

**Pull request merged**